src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            elements = grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index != 0 && size < index) {
+            throw new ArrayListIndexOutOfBoundsException("Index does not exist");
+        }
+        if (size == elements.length) {
+            elements = grow();
+        }
+        System.arraycopy(elements, index, elements, index + 1, elements.length - index - 1);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (size < index + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index does not exist");
+        }
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || size < index + 1) {
+            throw new ArrayListIndexOutOfBoundsException("Index does not exist");
+        }
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (size < index + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index does not exist");
+        }
```

Comment:
create separate private method for it.
```suggestion
        checkIndex(index);
```